src/main/java/com/google/googleidentity/test/TestInMemoryUserDetailsServiceProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.test;
```

Comment:
nit: it's usually helpful to keep test code closer to the code it helps test. Otherwise, we'll have tons of test code all in one spot as the codebase grows. Consider com.google.googleidentity.user.testing